numbat-wasm/www/service-worker.js

```diff
@@ -0,0 +1,181 @@
+// The prefix is necessary if multiple web apps will be hosted at the same
+// origin, e.g., as would be done on GitHub Pages. This prevents separate apps
+// from clashing when it comes to origin-scoped resources such as Caches, OPFS,
+// etc.
+const APP_PREFIX = "numbat_";
+// This should be updated any time cached assets change. It allows the entire
+// set of precached files to be refreshed atomically on SW update.
+// TODO: It would be nice if we did this automatically whenever a commit affects
```

Comment:
I guess this could be done in the deploy or build script, but it's not clear under what conditions exactly we should generate a new version. I think in that case we would want a single source of truth for all self-hosted artifacts (and remote URLs), such as a local JSON file. Then the local script could check the file hashes for changes and the service worker could fetch that JSON to determine the lists of files.

Reply:
I mean.. unless we change something completely unrelated like the docs, which are currently not part of this PWA, almost everything would require a VERSION_UUID change, right?

Could we update every time we use the `deploy.sh` script to push a new version/state to numbat.dev?

Reply:
I was originally thinking that we could put this into the deploy script, but the more I think about it the more sense it makes to use deterministic hashing of all built artifacts (i.e., anything that would get pushed via the deploy script). See my other comment in the main thread.